client/state/reader/conversations/reducer.js

```diff
@@ -0,0 +1,62 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_CONVERSATION_FOLLOW,
+	READER_CONVERSATION_MUTE,
+	READER_CONVERSATION_UPDATE_FOLLOW_STATUS,
+} from 'state/action-types';
+import {
+	CONVERSATION_FOLLOW_STATUS_FOLLOWING,
+	CONVERSATION_FOLLOW_STATUS_MUTING,
+} from './follow-status';
+import { combineReducers, createReducer, isValidStateWithSchema } from 'state/utils';
+import { itemsSchema } from './schema';
+import { key } from './utils';
+
+/**
+ * Tracks all known conversation following statuses.
+ */
+export const items = createReducer(
+	{},
+	{
+		[ READER_CONVERSATION_FOLLOW ]: ( state, action ) => {
+			const newState = assign( {}, state, {
+				[ key(
+					action.payload.siteId,
+					action.payload.postId
+				) ]: CONVERSATION_FOLLOW_STATUS_FOLLOWING,
+			} );
+			return newState;
+		},
+		[ READER_CONVERSATION_MUTE ]: ( state, action ) => {
+			const newState = assign( {}, state, {
+				[ key( action.payload.siteId, action.payload.postId ) ]: CONVERSATION_FOLLOW_STATUS_MUTING,
+			} );
+			return newState;
+		},
+		[ READER_CONVERSATION_UPDATE_FOLLOW_STATUS ]: ( state, action ) => {
+			// Use of a valid follow status is enforced by the schema
+			const newState = assign( {}, state, {
+				[ key( action.payload.siteId, action.payload.postId ) ]: action.payload.followStatus,
+			} );
+
+			if ( ! isValidStateWithSchema( newState, itemsSchema ) ) {
```

Comment:
My initial expectation here was that the update would automagically fail if schema validation failed, because I've provided `itemsSchema` to `createReducer`. 

If there's a way to do this without explicitly using `isValidStateWithSchema`, I'm all ears.

Reply:
what are you trying to guard against?

Reply:
Just trying to prevent invalid values of followStatus being set (only 'M', 'F' and null are acceptable). 

I can't find any other examples of validation being done in this way, so happy to remove it.